inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp

```diff
@@ -170,57 +167,87 @@ void MKLDNNIfNode::initSupportedPrimitiveDescriptors() {
     supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
-
 void MKLDNNIfNode::createPrimitive() {
-    const auto& eng = getEngine();
-
-    for (auto& map_rule : thenInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemThen[map_rule.to];
-
-        beforeThenMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
     }
+}
 
-    for (auto& map_rule : elseInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemElse[map_rule.to];
+bool MKLDNNIfNode::needPrepareParams() const {
+    return MKLDNNNode::needPrepareParams() || condition != (reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
+}
 
-        beforeElseMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+void MKLDNNIfNode::prepareParams() {
+    const auto& eng = getEngine();
+
+    if (isDynamic) {
+        condition = (reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
```

Comment:
as far as I know we doesn't support `bool` inside plugin
better to create config for condition with `U8` precision and interpret as `uint8_t` here

Reply:
but could you fix it?

Reply:
Fixed, thanks